docs/git/index.md

```diff
@@ -0,0 +1,220 @@
+---
+id: index
+title: Introduction
+sidebar_label: Introduction
+---
+
+Most software engineers use a tool called Git to organise their code, and collaborate with other people.
+
+Git can be used for many things, and we will see lots of them throughout the course.
+
+Each week, you're going to use Git to get a copy of your homework exercises, to submit your solutions, and to get feedback on your solutions. Today, you're going to learn how to do these things.
+
+## Learning Objectives
+
+By the end of this class, you should be able to:
+
+* Get a copy of your homework questions onto your computer.
+  * (To _clone_ a repository).
+* Save your answers, and send them to our volunteers.
+  * (To _commit_ and _push_ changes).
+* See and respond to feedback on your homework.
+  * (To _fork_ a repository, use Github's _pull request_ workflows, and _push_ further changes).
+* Explore how a file has changed over time.
+  * (To use Git's _history log_).
+
+---
+
+## Prerequisites
+
+1. A Github account. If you don't have one already, go to https://github.com/ and sign up for one.
+2. Installed Github Desktop. If you don't have it yet, go to https://desktop.github.com/ and follow its instructions to install it.
+3. Sign in to Github Desktop with your Github account. Run Github Desktop (TODO: Instructions on how to find it), click "Sign in to GitHub.com", and go through the steps to sign in.
+4. Installed VS Code.
+
+## What problem does Git solve?
+
+Git is used to solve many problems, but today we're going to focus on how it:
+* Helps us to share information.
+* Enables people to make their own changes to that information, and share it back.
+* Allows us to track how information has changed over time.
+
+## How does Git do that?
+
+Imagine your teacher wanted everyone in the class to answer three questions and to send back the answers. What capabilities would we need in order to be able to do that?
+
+**First**, the teacher needs to be able to write the questions, and **store** them somewhere.
+
+When we're using Git, we write things down in files in a folder (these could be text files, Word documents, images, or really any kind of file). When we save these files, we We say we're _committing_ them, and we call the folder where we're saving them a _repository_ (or _repo_ for short).
```

Comment:
From @40thieves:

You have the section near the end about commit messages, but I think we may need some additional explanation here on the difference between saving and committing. Basically to answer: why don't we commit on every save?

It may be hard to avoid getting into "what a good commit looks like", which I think we should avoid at this point.

Reply:
Yeah, avoiding "what does a good commit look like" is what led to me being really vague here... But maybe if we're explicitly teaching commits as checkpoints, and teaching discard-uncommitted-changes, we can use that as a useful framework... "Commit when you want to be able to see what you did before" is a reasonable mantra, without getting into being more precise/careful than that...

Reply:
> But maybe if we're explicitly teaching commits as checkpoints, and teaching discard-uncommitted-changes, we can use that as a useful framework... "Commit when you want to be able to see what you did before" is a reasonable mantra

I really like that idea. It feels like it would encourage diff inspection too.

I was thinking about proposing "1 exercise = 1 commit". But having thought about it, we don't want students to prioritise fixing commits over submitting work. Your softer guideline works better I think.